e2etest/newe2e_task_runazcopy.go

```diff
@@ -44,6 +51,35 @@ type AzCopyEnvironment struct {
 	InheritEnvironment bool
 }
 
+func (c *AzCopyCommand) applyTargetAuth(a Asserter, target ResourceManager) string {
+	intendedAuthType := EExplicitCredentialType.SASToken()
+	if tgt, ok := target.(AzCopyTarget); ok {
+		count := tgt.AuthType.Count()
+		a.AssertNow("target auth type must be single", Equal{}, count <= 1, true)
+		if count == 1 {
+			intendedAuthType = tgt.AuthType
+		}
+	} else if target.Location() == common.ELocation.S3() {
+		intendedAuthType = EExplicitCredentialType.S3()
+	} else if target.Location() == common.ELocation.GCP() {
+		intendedAuthType = EExplicitCredentialType.GCP()
+	}
+
+	switch intendedAuthType {
+	case EExplicitCredentialType.PublicAuth(), EExplicitCredentialType.None():
+		return target.URI(a, false)
+	case EExplicitCredentialType.SASToken():
+		return target.URI(a, true)
+	case EExplicitCredentialType.OAuth():
+		c.Environment.ServicePrincipalAppID = &GlobalConfig.E2EAuthConfig.SubscriptionLoginInfo.ApplicationID
```

Comment:
Does this force a specific type of OAuth?

Reply:
Yes, but this can now be adjusted with specifying env vars directly.